lib/view_model/hike_screen_model.dart

```diff
@@ -1,58 +1,340 @@
-    address = addresses.first.addressLine;
-    setState(ViewState.idle);
+    await databaseFunctions.fetchBeaconInfo(beacon.id).then((value) {
+      beacon = value;
+      isBeaconExpired = DateTime.fromMillisecondsSinceEpoch(beacon.expiresAt)
+          .isBefore(DateTime.now());
+      hikers.add(value.leader);
+      for (var i in value.followers) {
+        if (!followerId.contains(i.id)) {
+          hikers.add(i);
+          followerId.add(i.id);
+        }
+      }
+      var lat = double.parse(value.location.lat);
+      var lon = double.parse(value.location.lon);
+      route.add(LatLng(lat, lon));
+      address = addresses.first.addressLine;
+      markers.add(Marker(
+        markerId: MarkerId("0"),
+        position: route.first,
+        infoWindow: InfoWindow(
+          title: 'Initial Location',
+        ),
+      ));
+      markers.add(Marker(
+        markerId: MarkerId("1"),
+        position: route.last,
+        infoWindow: InfoWindow(
+          title: 'Current Location',
+        ),
+      ));
+      for (var i in value.landmarks) {
+        markers.add(Marker(
+          markerId: MarkerId((markers.length + 1).toString()),
+          position: LatLng(
+              double.parse(i.location.lat), double.parse(i.location.lon)),
+          infoWindow: InfoWindow(
+            title: '${i.title}',
+          ),
+          icon: BitmapDescriptor.defaultMarkerWithHue(BitmapDescriptor.hueBlue),
+        ));
+      }
+      // for (var i in value.followers) {
+      //   markers.add(Marker(
+      //     markerId: MarkerId((markers.length + 1).toString()),
+      //     position: LatLng(
+      //         double.parse(i.location.lat), double.parse(i.location.lon)),
+      //     infoWindow: InfoWindow(
+      //       title: '${i.name}',
+      //     ),
+      //     icon: BitmapDescriptor.defaultMarkerWithHue(
+      //         BitmapDescriptor.hueYellow),
+      //   ));
+      // }
+      //notifyListeners();
+    });
+  }
+
+  Future<void> setupSubscriptions(bool isExpired) async {
+    if (isBeaconExpired || isExpired) return;
+    if (isLeader) {
+      // distanceFilter (in m) can be changed to reduce the backend calls
+      await loc.changeSettings(interval: 3000, distanceFilter: 0.0);
+      _leaderLocation =
+          loc.onLocationChanged.listen((LocationData currentLocation) async {
+        if (DateTime.fromMillisecondsSinceEpoch(beacon.expiresAt)
+            .isBefore(DateTime.now())) _leaderLocation.cancel();
+        Coordinates coordinates =
+            Coordinates(currentLocation.latitude, currentLocation.longitude);
+        var addresses =
+            await Geocoder.local.findAddressesFromCoordinates(coordinates);
+
+        String _address = addresses.first.addressLine;
+        if (true) {
+          databaseFunctions.init();
+          await databaseFunctions.updateLeaderLoc(beacon.id,
+              LatLng(currentLocation.latitude, currentLocation.longitude));
+
```

Comment:
Here, updated location is not getting added the `route` list because of which [this](https://github.com/CCExtractor/beacon/pull/51#pullrequestreview-850656079) issue is caused

> It zooms into the current location because of which instead of the entire route being shared, just that specific marker is shown in the screenshot

Reply:
Sure would fix that 👍

Reply:
I have fixed it 👍